cpp/modmesh/buffer/pymod/wrap_SimpleArrayPlex.cpp

```diff
@@ -117,6 +157,87 @@ static void verify_python_value_datatype(pybind11::object const & value, DataTyp
     }
 }
 
+// NOLINTNEXTLINE(misc-use-anonymous-namespace)
+static modmesh::detail::shape_type make_shape(pybind11::object const & shape_in)
+{
+    modmesh::detail::shape_type shape;
+    try
+    {
+        shape.push_back(shape_in.cast<size_t>());
+    }
+    catch (const pybind11::cast_error &)
+    {
+        shape = shape_in.cast<std::vector<size_t>>();
+    }
+    return shape;
+}
+
+/// Get the typed array value by the key
+/// @tparam T the type of the key
+template <typename T>
+// NOLINTNEXTLINE(misc-use-anonymous-namespace)
+static pybind11::object get_typed_array_value(const SimpleArrayPlex & array_plex, T key)
+{
+#define DECL_MM_GET_TYPED_ARRAY_VALUE_BY_INDEX(DataType, ArrayType)                     \
+    case DataType:                                                                      \
+    {                                                                                   \
+        const auto * array = static_cast<const ArrayType *>(array_plex.instance_ptr()); \
+        return pybind11::cast(array->at(key));                                          \
```

Comment:
At the beginning, I tried:
```cpp
[](wrapped_type & self, ssize_t key)
{ 
return execute_callback_with_typed_array(
    self,
    [](auto & array)
    { return array.at(key); }); 
})
```

However, the compiler cannot deduce the correct type. After spending some time trying to play around with template tricks and failing, I think it isn't worth using `execute_callback_with_typed_array` here.

Reply:
A macro trick is fine here.  We can keep it for a while and decide how to improve, if needed.